packages/database/src/migrations/20201009061046-UpdateLaosEocEntities-modifies-data.js

```diff
@@ -0,0 +1,257 @@
+      lon: 102.63719558715822,
+    },
+  },
+  'Vientiane Province': {
+    capital: {
+      name: 'Phônhông',
+      lat: 18.5,
+      lon: 102.416667,
+    },
+  },
+  Xiangkhouang: {
+    capital: {
+      name: 'Pek',
+      lat: 19.416667,
+      lon: 103.166667,
+    },
+  },
+  Xaisomboun: {
+    capital: {
+      name: 'Anouvong',
+      lat: 18.9057,
+      lon: 103.092,
+    },
+  },
+};
+
+// Pre-calculate entity codes (needed for both up() and down()), and id
+for (const provinceName of Object.keys(PROVINCES)) {
+  const { capital } = PROVINCES[provinceName];
+  const uppercaseName = capital.name.toUpperCase().replace(' ', '_');
+  PROVINCES[provinceName].capital.code = `LA_PROV_CAPITAL_${uppercaseName}`;
+  PROVINCES[provinceName].capital.id = generateId();
+}
+
+const getLaosCountryEntityId = async function (db) {
+  const results = await db.runSql(`SELECT id FROM entity WHERE code = 'LA';`);
+
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+
+  throw new Error('Entity not found');
+};
+
+const getEntityHierarchyId = async function (db) {
+  const results = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = 'laos_eoc';`);
+
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+
+  throw new Error('Entity hierarchy not found');
+};
+
+const clearData = async function (db) {
+  await db.runSql(
+    "DELETE FROM entity_relation where child_id IN (SELECT id FROM entity WHERE code like 'LA_PROV_CAPITAL_%');",
+  );
+  await db.runSql("DELETE FROM entity WHERE code like 'LA_PROV_CAPITAL_%';");
+};
+
+exports.up = async function (db) {
+  const parentId = await getLaosCountryEntityId(db);
+
+  const entityHierarchyId = await getEntityHierarchyId(db);
+
+  // Delete old data
+  await clearData(db);
+
+  /*
+   * Entity
+   */
+  let sql =
+    'INSERT INTO entity (id, code, parent_id, name, type, point, country_code, bounds, metadata) VALUES \n';
+
+  const rowsSql = [];
+
+  for (const province of Object.values(PROVINCES)) {
+    let rowSql = '(';
+
+    const point = { type: 'Point', coordinates: [province.capital.lon, province.capital.lat] };
+
+    rowSql +=
+      `'${province.capital.id}', ` +
+      `'${province.capital.code}', ` +
+      `'${parentId}', ` +
+      `'${province.capital.name}', ` +
+      `'city', ` +
+      `ST_Force2D(ST_GeomFromGeoJSON('${JSON.stringify(point)}')), ` +
+      `'LA', ` +
+      `ST_Expand(ST_Envelope(ST_GeomFromGeoJSON('${JSON.stringify(point)}')::geometry), 1), ` +
+      `'{}'`;
+
+    rowSql += ')';
+
+    rowsSql.push(rowSql);
+  }
+
+  sql += rowsSql.join(',\n');
+
+  sql += ';';
+
+  await db.runSql(sql);
+
+  /*
+   * Entity relation
+   */
+  for (const province of Object.values(PROVINCES)) {
+    await insertObject(db, 'entity_relation', {
+      id: generateId(),
+      parent_id: parentId,
+      child_id: province.capital.id,
+      entity_hierarchy_id: entityHierarchyId,
+    });
+  }
+
+  return null;
```

Comment:
Is there a reason to return null here? Should I start doing it? Otherwise perhaps this is unneeded?

Reply:
Hmm, I just started copying other people doing it, I don't think it matters at all